app/models/goal.py

```diff
@@ -3,4 +3,12 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    tasks = db.relationship('Task', lazy=True)
```

Comment:
This db.relationship should generate a list of tasks (Task models) but it may need `back_populates` added:
```suggestion
    tasks = db.relationship('Task', back_populates="goal", lazy=True)
```
When using `back_populates` I think it's needed in the db.relationship on both ends (one & many).  However, using `backref` simplifies this:
```suggestion
    tasks = db.relationship('Task', backref="goal", lazy=True)
```
When using `backref`, the relationship on the other side is generated automatically, so line 12 of task.py can be removed.  Each Goal will have a `tasks` var that is a list of Task models, and each Task model will have a var `goal` that is the `Goal` model with the correct goal_id.

Here's a link to an explanation using SQLAlchemy syntax (the syntax here is slightly different because we're using the Flask-SQLAlchemy layer: https://backend-course.cornellappdev.com/cheat-sheets/concept-faqs).